include/data/data.hpp

```diff
@@ -50,6 +51,27 @@ struct Event
      */
     static Event fromJson(nlohmann::json j);
 
+    /**
+     * @brief Encode this event object using bech32 note encoding
+     *
+     * @return std::string containing the encoded note string
+     */
+    std::string toBech32Note();
+
+    /**
+     * @brief Encode this event object using bech32 naddr encoding
+     *
+     * @return std::string containing the encoded naddr string
+     */
+    std::string toBech32Naddr();
+
+    /**
+     * @brief Encode this event object using bech32 nevent encoding
+     *
+     * @return std::string containing the encoded nevent string
+     */
+    std::string toBech32Nevent();
```

Comment:
More idiomatic C++ would define these as templated variants of a single `encode()` function, such that some invocation like `encode<Bech32NEvent>()` would return the event encoded in `nevent` format.  We may wish to refactor the code to use such idioms in the future, as we reorganize the code around the new structure (as defined in the in-progress LucidChart diagram I've shared on Slack).

That said, we don't need to worry about that in this PR; I'm just putting the idea out there.